### PR TITLE
Rename extension to .tsx if JSX is present

### PR DESCRIPTION
**Pull request by skovhus**
Thanks for a great codemod! 

When a file contains JSX it is now renamed to .tsx instead of .ts. See https://www.typescriptlang.org/docs/handbook/jsx.html

Besides, I have fixed these smaller things while testing:
- Correctly parse CLI `printWidth` as number (currently it fails if you try it using that option in the CLI)
- Update out-dated yarn.lock (installing the project showed some inconsistencies in the yarn.lock file)

### CODE REVIEW

**Comment by skovhus** [comment]
@kevinbarabash @vfedyk-lohika anything I can do to land this?

File: src/cli.js [hunk 1]
```diff
@@ -78,7 +82,8 @@ const cli = argv => {
       const outCode = convert(inCode, options);
 
       if (program.write) {
-        const outFile = file.replace(/\.js$/, ".ts");
+        const extension = detectJsx(outCode) ? ".tsx" : ".ts";
```
Comment (lautis) [inline src/cli.js]:
```suggestion
        const extension = detectJsx(inCode) ? ".tsx" : ".ts";
```

The detectJsx function is using babel settings for flow, and does not support TypeScript-specific syntax.

Comment (skovhus) [reply src/cli.js]: Ah, that makes sense. 👍

**Comment by k4b7** [comment]
[clabot:check]

File: src/detect-jsx.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const { parse } = require("@babel/parser");
+const traverse = require("../babel-traverse/lib/index.js").default;
+
+const { parseOptions } = require("./convert.js");
+
+const detectJsx = tsCode => {
```
Comment (k4b7) [inline src/detect-jsx.js]: `detectJsx` is called with flow code.  Eventually I'd like this project to do conversions both ways so I think renaming `tsCode` to `code` is appropriate.

File: src/detect-jsx.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  let jsx = false;
+  const ast = parse(tsCode, parseOptions);
```
Comment (k4b7) [inline src/detect-jsx.js]: It would be nice to avoid extra calls to `parse` but that could be a separate PR since changes will need to be made elsewhere.

**Review by k4b7 — APPROVED** [review]
LGTM.  One minor nit about the param name for `detectJsx`.

**Comment by skovhus** [comment]
[clabot:check]
